### PR TITLE
Resolve #1764

### LINKED ISSUE #1764: Replace and Nulls
It's worth noting that Set-ADUser with a replace operation will generate an error if any of the properties to update are being updated will a null. An error will be generated and the command will fail even if several properties to be updated have values. For example, if $CSV.desc = null and $CSV.dept = "D602",  the following will fail with an ADInvalidOperationException:

$account | Set-ADUser -Replace @{department=$($CSV.dept);description=$($CSV.desc)} -Credential $cred

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8918842f-aade-158a-b4bc-6b78cf9baf6b
* Version Independent ID: 4ea95b5f-9e2a-8ab2-95b6-3d8524b712e3
* Content: [Set-ADUser (addsadministration)](https://docs.microsoft.com/en-us/powershell/module/addsadministration/set-aduser?view=win10-ps)
* Content Source: [docset/windows/addsadministration/Set-ADUser.md](https://github.com/MicrosoftDocs/windows-powershell-docs/blob/live/docset/windows/addsadministration/Set-ADUser.md)
* Product: **w10**
* Technology: **powershell-windows**
* GitHub Login: @andreabarr
* Microsoft Alias: **v-anbarr**

### PR DESCRIPTION
**Pull request by velkovb**
Addition to parameter description.
https://github.com/MicrosoftDocs/windows-powershell-docs/issues/1764

### CODE REVIEW

**Comment by e0i** [comment]
cc @dariomws @get-itips @tseward

@JohanFreelancer9

File: docset/windows/addsadministration/Set-ADUser.md [hunk 1]
```diff
@@ -199,7 +199,7 @@ Accept wildcard characters: False
 Specifies values to add to an object property.
 Use this parameter to add one or more values to a property that cannot be modified using a cmdlet parameter.
 To modify an object property, you must use the LDAP display name.
-You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon.
+You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties has null or empty value the cmdlet will fail.
```
Comment (tseward) [inline docset/windows/addsadministration/Set-ADUser.md:202]: I would rephrase this to "the cmdlet will return an error.".

Comment (JohanFreelancer9) [inline docset/windows/addsadministration/Set-ADUser.md:202]:
```suggestion
You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties have a null or empty value the cmdlet will return an error.
```

File: docset/windows/addsadministration/Set-ADUser.md [hunk 2]
```diff
@@ -1185,7 +1185,7 @@ Accept wildcard characters: False
 Specifies that the cmdlet remove values of an object property.
 Use this parameter to remove one or more values of a property that cannot be modified using a cmdlet parameter.
 To remove an object property, you must use the LDAP display name.
-You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon.
+You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties has null or empty value the cmdlet will fail.
```
Comment (JohanFreelancer9) [inline docset/windows/addsadministration/Set-ADUser.md:1188]:
```suggestion
You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties have a null or empty value the cmdlet will return an error.
```

File: docset/windows/addsadministration/Set-ADUser.md [hunk 3]
```diff
@@ -1213,7 +1213,7 @@ Accept wildcard characters: False
 Specifies values for an object property that will replace the current values.
 Use this parameter to replace one or more values of a property that cannot be modified using a cmdlet parameter.
 To modify an object property, you must use the LDAP display name.
-You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon.
+You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties has null or empty value the cmdlet will fail.
```
Comment (JohanFreelancer9) [inline docset/windows/addsadministration/Set-ADUser.md:1216]:
```suggestion
You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties have a null or empty value the cmdlet will return an error.
```

File: docset/winserver2012r2-ps/addsadministration/Set-ADUser.md [hunk 4]
```diff
@@ -198,7 +198,7 @@ Accept wildcard characters: False
 Specifies values to add to an object property.
 Use this parameter to add one or more values to a property that cannot be modified using a cmdlet parameter.
 To modify an object property, you must use the LDAP display name.
-You can specify multiple values to a property by specifying a comma-separated list of values and more than one property by separating them using a semicolon.
+You can specify multiple values to a property by specifying a comma-separated list of values and more than one property by separating them using a semicolon. If any of the properties has null or empty value the cmdlet will fail.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/addsadministration/Set-ADUser.md:201]:
```suggestion
You can specify multiple values to a property by specifying a comma-separated list of values and more than one property by separating them using a semicolon. If any of the properties have a null or empty value the cmdlet will return an error.
```

File: docset/winserver2012r2-ps/addsadministration/Set-ADUser.md [hunk 5]
```diff
@@ -1184,12 +1184,12 @@ Accept wildcard characters: False
 Specifies that the cmdlet remove values of an object property.
 Use this parameter to remove one or more values of a property that cannot be modified using a cmdlet parameter.
 To remove an object property, you must use the LDAP display name.
-You can remove more than one property by specifying a semicolon-separated list.
+You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties has null or empty value the cmdlet will fail.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/addsadministration/Set-ADUser.md:1187]:
```suggestion
You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties have a null or empty value the cmdlet will return an error.
```

File: docset/winserver2012r2-ps/addsadministration/Set-ADUser.md [hunk 6]
```diff
@@ -1212,7 +1212,7 @@ Accept wildcard characters: False
 Specifies values for an object property that will replace the current values.
 Use this parameter to replace one or more values of a property that cannot be modified using a cmdlet parameter.
 To modify an object property, you must use the LDAP display name.
-You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon.
+You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties has null or empty value the cmdlet will fail.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/addsadministration/Set-ADUser.md:1215]:
```suggestion
You can specify multiple values to a property by specifying a comma-separated list of values, and more than one property by separating them using a semicolon. If any of the properties have a null or empty value the cmdlet will return an error.
```

File: docset/winserver2012r2-ps/addsadministration/Set-ADUser.md [hunk 7]
```diff
@@ -1311,10 +1311,9 @@ Accept wildcard characters: False
 ```
 
 ### -ServicePrincipalNames
-Specifies the service principal names for the account.
-This parameter sets the **ServicePrincipalNames** property of the account.
-The LDAP display name (**ldapDisplayName**) for this property is servicePrincipalName.
-This parameter uses the following syntax to add remove, replace, or clear service principal name values:
+Specifies the service principal names for the account. This parameter sets the ServicePrincipalNames property of the account. The LDAP display name (ldapDisplayName) for this property is servicePrincipalName. This parameter uses the following syntax to add, remove, replace or clear service principal name values.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/addsadministration/Set-ADUser.md:1314]:
```suggestion
Specifies the service principal names for the account. This parameter sets the ServicePrincipalNames property of the account. The LDAP display name (ldapDisplayName) for this property is servicePrincipalName. This parameter uses the following syntax to add, remove, replace, or clear service principal name values.
```

**Comment by get-itips** [comment]
I am ok once @tseward suggestion is applied.
